<launch>
	<node name="planner_service" pkg="multiagent_planning" type="planner_node" required="true"/>

	<node name="rviz_visualizer" pkg="rviz" type="rviz"/>

	<group ns="agent1">
		<node name="agent" pkg="multiagent_planning" type="agent_node" args="1 2 0 90" required="true">
			<remap from="/agent_feedback" to="/agent1/agent_feedback"/>
			<remap from="/update_goal" to="/agent1/update_goal"/>
		</node>
	</group>

	<group ns="agent2">
		<node name="agent" pkg="multiagent_planning" type="agent_node" args="2 0 2 0" required="true">
			<remap from="/agent_feedback" to="/agent2/agent_feedback"/>
			<remap from="/update_goal" to="/agent2/update_goal"/>
		</node>
	</group>

	<group ns="client1">
		<node name="update_goal_client" pkg="multiagent_planning" type="update_goal_client" args="1 2 5 0">
			<remap from="/update_goal" to="/agent1/update_goal"/>
			<remap from="/planned_path" to="/client1/planned_path"/>
		</node>
	</group>

	<group ns="client2">
		<node name="update_goal_client" pkg="multiagent_planning" type="update_goal_client" args="2 6 2 0">
			<remap from="/update_goal" to="/agent2/update_goal"/>
			<remap from="/planned_path" to="/client2/planned_path"/>
		</node>
	</group>
</launch>